src/main/java/com/amazon/ion/impl/macro/IonReaderFromReaderAdapter.kt

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.macro
+
+import com.amazon.ion.*
+import java.lang.UnsupportedOperationException
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.util.*
+
+/**
+ * An [IonReader] that delegates to a [ReaderAdapter].
+ */
+internal class IonReaderFromReaderAdapter(val reader: ReaderAdapter) : IonReader {
```

Comment:
This was the easiest way to be able to provide an `IonReaderContinuableCore` (which doesn't implement `IonReader`, but does have a `ReaderAdapter`) to `IonWriter.writeValue(IonReader)`.